### ISSUE TITLE
BaseModelView errors on calculated column not existing in parent table.

### ISSUE THREAD

**Issue by wintersandroid** [Raizlabs/DBFlow#152]
I have a tablemodel defined as 
```
@Table(value = "WALLET_ORDERS")
public class WalletOrdersTable extends BaseModel {
  @Column(columnType = Column.PRIMARY_KEY_AUTO_INCREMENT)
  public long _id;

  @Column(name = "ACCOUNT_ID", notNull = true)
  public Long accountId;

  @Column(name = "PAYMENT_DATE", notNull = true)
  public String paymentDateStr;

  @Column(name = "STATUS", notNull = true)
  public String status;

  @Column(name = "ORDERID", notNull = true)
  public String order_id;

  @Column(name = "DETAILS")
  public String details;

  @Column(name = "SUBSCRIPTION")
  public String subscription;

  @Column(name = "TOTAL", notNull = true)
  public String total;

  @Column(name = "PAID_AS", notNull = true)
  public String paidAs;

}

```

and a modelview on that table.

```
@ModelView(name="WALLET_ORDERS_ORDERED",query=
    " select *, " +
        "   case status " +
        "     when 'chargeable' then 1 " +
        "     when 'charged' then 2 " +
        "     when 'cancelled' then 3 " +
        "     else 4 " +
        "   end as STATUS_ORDER " +
        " from WALLET_ORDERS " +
        " order by PAYMENT_DATE desc "
    )
public class WalletOrdersOrderedView extends BaseModelView<WalletOrdersTable> {
  @Column
  public long _id;

  @Column(name = "ACCOUNT_ID", notNull = true)
  public Long accountId;

  @Column(name = "PAYMENT_DATE", notNull = true)
  public String paymentDateStr;

  @Column(name = "STATUS", notNull = true)
  public String status;

  @Column(name = "ORDERID", notNull = true)
  public String order_id;

  @Column(name = "DETAILS")
  public String details;

  @Column(name = "SUBSCRIPTION")
  public String subscription;

  @Column(name = "TOTAL", notNull = true)
  public String total;

  @Column(name = "PAID_AS", notNull = true)
  public String paidAs;

  @Column(name="STATUS_ORDER")
  public int statusOrder;
}
```

generates an error on the following in WalletOrdersOrderedView$View.java
```
 public ConditionQueryBuilder<WalletOrdersOrderedView> getPrimaryModelWhere(WalletOrdersOrderedView model) {
    return new ConditionQueryBuilder<WalletOrdersOrderedView>(WalletOrdersOrderedView.class, Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table._ID).is(((long)model._id)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.ACCOUNT_ID).is(((java.lang.Long)model.accountId)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.PAYMENT_DATE).is(((java.lang.String)model.paymentDateStr)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.STATUS).is(((java.lang.String)model.status)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.ORDERID).is(((java.lang.String)model.order_id)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.DETAILS).is(((java.lang.String)model.details)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.SUBSCRIPTION).is(((java.lang.String)model.subscription)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.TOTAL).is(((java.lang.String)model.total)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.PAID_AS).is(((java.lang.String)model.paidAs)),Condition.column(nz.org.winters.android.merchantwidget.database.WalletOrdersTable$Table.STATUS_ORDER).is(((int)model.statusOrder)));
  }

```

The error being Error:(68, 1395) error: cannot find symbol variable STATUS_ORDER
where its looking for STATUS_ORDER on the original table, it should be added to the View as a define there.

**Comment by agrosner** [comment]
interesting. I will look into this within the next couple days.

**Comment by wintersandroid** [comment]
I would be inclined (as i thought it would) to make a viewmodel actually
create a database view and use that as if it was a table, which is the
method I have chosen after trying the model view...

:)


On 20 March 2015 at 03:29, Andrew G <notifications@github.com> wrote:

> interesting. I will look into this within the next couple days.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Raizlabs/DBFlow/issues/152#issuecomment-83607591>.
>



-- 
*Mathew Winters*
*See my other apps*
*https://play.google.com/store/apps/developer?id=Mathew+Winters
<https://play.google.com/store/apps/developer?id=Mathew+Winters>*

**Comment by agrosner** [comment]
fixed in develop for 1.8.0
